### ISSUE TITLE
CoverArtController reimplementation

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#1376]
The cover art acquisition process has some design bugs. Requires overall reimplementation and testing.

 - The exception handling needs to be rewritten. It is `one of the Throwable catch implementations done in Legacy`. 
 - There are no unit tests needed for the Extreme Development. We often see glitches happening, it ’s difficult to improve.
   - At least we need to add a test.
   - If the spec are incomplete, another issue may be issued.
   - The main review point of view will be the functional unit of CoverArtController, but the scope of modification will affect the parser package.

<details open>
<summary>Fixes</summary>

 - CoverArtController#[CoverArtRequest](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/controller/CoverArtController.java#L363)
 - CoverArtController#[sendFallback](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/controller/CoverArtController.java#L207-L222) `#1404`
  - CoverArtController#[getImageInputStream](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/controller/CoverArtController.java#L280-L282)
     - CoverArtController#[getImageInputStreamWithType](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/controller/CoverArtController.java#L284-L309) `#1387`
        - JaudiotaggerParser#[getArtwork](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/service/metadata/JaudiotaggerParser.java#L279-L289)  `#1392`
     - CoverArtController#[getImageInputStreamForVideo](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/controller/CoverArtController.java#L311-L317) `#1398`
     - CoverArtController#[sendFallback](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/controller/CoverArtController.java#L207-L222) `#1400`
 - MediaFileService#[createMediaFile](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java#L473-L557)
   - MediaFileService#[findCoverArt](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java#L615-L631)
     - JaudiotaggerParser#[getArtwork](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/service/metadata/JaudiotaggerParser.java#L279-L289) `#1392`

Sub issues  :

 - Whether it is an implementation considering the case where the embedded image cannot be acquired by JaudiotaggerParser#[getArtwork](https://github.com/airsonic/airsonic/blob/5ccca059d5cfe3dd19734c27861b434ea21b43d8/airsonic-main/src/main/java/org/airsonic/player/service/metadata/JaudiotaggerParser.java#L279-L289)
   - #1387
   - #1392
 - Appropriate processing of video image paths
   - #1398
 - If the image cannot be obtained (fallback)
   - #1400
   - #1404

</details>

**Comment by tesshucom** [comment]
Most noticeable issues have been fixed and coverage has exceeded 60%. The main branches are covered because it was a test addition to the part where the problem is occurring. This issues will be closed. Any further issues will be addressed in another new issue. 

The points to worry about are that the semaphore probably won't work, and that creating thumbnails for the video is little expensive (But in a good way it's balanced now, because semaphore size=4 is too big). This needs to be resolved at the same time. There is no problem even if begin after #1373.

**Comment by tesshucom** [comment]
The lack of cases that cannot be obtained when reading embedded cover art. Such data cannot be registered before, but the cover art can be deleted after scanning a file with embedded cover art.
